checker/checker.go

```diff
@@ -323,46 +323,46 @@ func (c *checker) checkBlockStmt(scope *parser.Scope, typ parser.ObjType, block
 		}
 
 		lookupByType := builtin.Lookup.ByType[typ]
-		if !foundSource {
-			// If the function is not a builtin, retrieve it from the scope and then
-			// type check it.
-			_, ok := lookupByType.Func[name]
-			if !ok {
-				obj := scope.Lookup(name)
-				if obj == nil {
-					return ErrFirstSource{call}
-				}
+		// If the function is not a builtin, retrieve it from the scope and then
+		// type check it.
+		_, ok := lookupByType.Func[name]
+		if !ok {
+			obj := scope.Lookup(name)
+			if obj == nil {
+				return ErrIdentNotDefined{Ident: call.Func.Ident}
+			}
 
-				// The retrieved object may be either a function declaration or a field
-				// in the current scope's function parameters.
-				var callType *parser.Type
-				switch obj.Kind {
-				case parser.DeclKind:
-					switch n := obj.Node.(type) {
-					case *parser.FuncDecl:
-						callType = n.Type
-					case *parser.AliasDecl:
-						callType = n.Func.Type
-					}
-				case parser.FieldKind:
-					field, ok := obj.Node.(*parser.Field)
-					if ok {
-						callType = field.Type
-					}
+			// The retrieved object may be either a function declaration or a field
+			// in the current scope's function parameters.
+			var callType *parser.Type
+			switch obj.Kind {
+			case parser.DeclKind:
+				switch n := obj.Node.(type) {
+				case *parser.FuncDecl:
+					callType = n.Type
```

Comment:
I think even though we don't support user defined methods atm, we can implement checking for function/method already.